### PR TITLE
Dev/wxng/more solutions

### PR DESCRIPTION
**Pull request by sharpeye096**

### CODE REVIEW

File: leetcode/stringsLinearList/726NumberofAtoms.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#pragma once
+#include <string>
+#include <map>
+#include <iostream>
+#include <sstream>
+using namespace std;
+
+class Solution {
+private:
+    map<string, int> processParentheses(const string& formula, int startIndex) {
```
Comment (WindTalkerGold) [inline leetcode/stringsLinearList/726NumberofAtoms.h:10]: method is long. Reduce it

File: leetcode/stringsLinearList/726NumberofAtoms.h [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        // startIndex is (
```
Comment (WindTalkerGold) [inline leetcode/stringsLinearList/726NumberofAtoms.h]: start Index is '('

File: leetcode/stringsLinearList/726NumberofAtoms.h [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        map<string, int> elementsInside;
+        int currentIndex = startIndex+1;
+        string currentAtom;
+        int currentAtomCount = 0;
+        while(currentIndex < formula.size()) {
+            char ch = formula[currentIndex];
+            if (ch == ')') {
```
Comment (WindTalkerGold) [inline leetcode/stringsLinearList/726NumberofAtoms.h:18]: logic seems to be complex, using state machine?

File: leetcode/stringsLinearList/726NumberofAtoms.h [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        }
+        //
+        elementsInside[")"] = currentIndex;
+        return elementsInside;
+    }
+
+
+public:
+    string countOfAtoms(const string& formula) {
+        string formula2 = formula + ")1";
```
Comment (WindTalkerGold) [inline leetcode/stringsLinearList/726NumberofAtoms.h:81]: seems ugly. Anyway to avoid this?

File: leetcode/stringsLinearList/726NumberofAtoms.h [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            else {
+                break;
+            }
+        }
+
+        for (auto pair : elementsInside) {
+            elementsInside[pair.first] = pair.second * elementsCount;
+        }
+        //
+        elementsInside[")"] = currentIndex;
```
Comment (WindTalkerGold) [inline leetcode/stringsLinearList/726NumberofAtoms.h:74]: Can't tell what's wrong. But the usage seems hacky

File: leetcode/stringsLinearList/726NumberofAtoms.h (hunk 2 above)
Comment (sharpeye096) [reply leetcode/stringsLinearList/726NumberofAtoms.h]: Ok

File: leetcode/stringsLinearList/726NumberofAtoms.h (hunk 3 above)
Comment (sharpeye096) [reply leetcode/stringsLinearList/726NumberofAtoms.h:18]: Overkill for a leetcode answer

File: leetcode/stringsLinearList/726NumberofAtoms.h (hunk 4 above)
Comment (sharpeye096) [reply leetcode/stringsLinearList/726NumberofAtoms.h:81]: Just to make the method above usable to the global cases without any change

File: leetcode/stringsLinearList/726NumberofAtoms.h (hunk 5 above)
Comment (sharpeye096) [reply leetcode/stringsLinearList/726NumberofAtoms.h:74]: Otherwise we will need a class

**Review by WindTalkerGold — APPROVED** [review]

**Pull request merged**